src/interfaces/interfaces.ts

```diff
@@ -226,18 +275,23 @@ namespace interfaces {
     applyMiddleware(...middleware: Middleware[]): void;
     snapshot(): void;
     restore(): void;
-    createChild(): Container;
+    createChild<C extends BindingMap = any>(containerOptions?: ContainerOptions): Container<C, T>;
   }
 
-  export type Bind = <T = unknown>(serviceIdentifier: ServiceIdentifier<T>) => BindingToSyntax<T>;
+  export type Bind<T extends BindingMap = any> =
+    <B extends ContainerBinding<T, K>, K extends ContainerIdentifier<T> = any>(serviceIdentifier: K) => BindingToSyntax<B>;
```

Comment:
Note, in the interest of keeping changes to a minimum, I haven't touched the `BindingToSyntax` type, but we may want to consider tweaking it — either as part of this change, or a future change — because:

```ts
container.bind('foo').toDynamicValue(
  ({container}) => container.get('bar') // not strongly typed
)
```